### PR TITLE
ofmcc-858 - update getLookup in backend, add Application Operation Cost

### PR DESCRIPTION
**Pull request by vietle-cgi**

### CODE REVIEW

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: frontend/src/components/applications/YearlyOperatingCost.vue [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+      totalNumberFormat: {
+        decimal: '.',
+        separator: ',',
+        prefix: '$ ',
+      },
+      model: {},
+      UTILITIES_INFO_TXT: 'This is a placeholder message',
+      FURNITURE_INFO_TXT: 'This is a placeholder message',
+      UPKEEP_INFO_TXT: 'This is a placeholder message',
+      MAINTENANCE_INFO_TXT: 'This is a placeholder message',
```
[the remark this answers is not in the document]
Comment (vietle-cgi) [reply frontend/src/components/applications/YearlyOperatingCost.vue]: it is fixed

File: frontend/src/stores/applications.js [hunk 2]
```diff
@@ -37,6 +37,10 @@ function checkStaffingComplete(application) {
   )
 }
 
+function checkOperatingCostsComplete(application) {
+  return application?.facilityType && application?.totalYearlyOperatingCosts + application?.totalYearlyFacilityCosts > 0
```
[the remark this answers is not in the document]
Comment (vietle-cgi) [reply frontend/src/stores/applications.js:41]: as discussed, I am unable to add parentheses...

File: frontend/src/utils/rules.js [hunk 3]
```diff
@@ -17,10 +17,10 @@ const rules = {
   YYYY: [(v) => (v > 1900 && v < 2100) || 'A valid year is required'],
   notRequired: [() => true],
   max(number) {
-    return (v) => !v || v <= number || 'Max exceeded'
+    return (v) => !v || v <= number || `Max exceeded: ${number.toLocaleString('en-us')}`
```
[the remark this answers is not in the document]
Comment (vietle-cgi) [reply frontend/src/utils/rules.js]: it is fixed

File: frontend/src/utils/rules.js [hunk 4]
```diff
@@ -17,10 +17,10 @@ const rules = {
...
   },
   min(number) {
-    return (v) => !v || v >= number || 'Min exceeded'
+    return (v) => !v || v >= number || `Min exceeded: ${number.toLocaleString('en-us')}`
```
[the remark this answers is not in the document]
Comment (vietle-cgi) [reply frontend/src/utils/rules.js]: it is fixed

File: frontend/src/views/applications/OperatingCostsView.vue [hunk 5]
```diff
@@ -88,5 +120,36 @@ export default {
...
+    updateModel(updatedModel) {
+      Object.entries(updatedModel)?.forEach(([key, value]) => {
+        this.costsModel[key] = Number(value)
+        this.model[key] = Number(value)
+      })
+    },
+  },
 }
 </script>
+<style scoped></style>
```
[the remark this answers is not in the document]
Comment (vietle-cgi) [reply frontend/src/views/applications/OperatingCostsView.vue]: it is removed

File: frontend/src/views/applications/OperatingCostsView.vue [hunk 6]
```diff
@@ -1,43 +1,59 @@
...
-        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris
-        nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident,
-        sunt in culpa qui officia deserunt mollit anim id est laborum.
-      </p>
-    </v-row>
-    <v-row no-gutters class="my-6">
-      <p>Donec iaculis nec quam vel congue. Fusce consequat mattis rhoncus. Sed id ipsum sed purus placerat euismod vel ut erat. Nullam ligula leo, fermentum vel interdum sit amet, tempor at nunc.</p>
-    </v-row>
+    <YearlyOperatingCost v-if="model.facilityType" :readonly="readonly" @update="updateModel" />
+    <YearlyFacilityCost v-if="model.facilityType" class="my-8" :readonly="readonly" :facilityType="model.facilityType" @update="updateModel" />
```
[the remark this answers is not in the document]
Comment (vietle-cgi) [reply frontend/src/views/applications/OperatingCostsView.vue]: it is updated

File: frontend/src/views/applications/OperatingCostsView.vue [hunk 7]
```diff
@@ -52,24 +68,40 @@ export default {
...
     },
   },
+  emits: ['process'],
   data() {
     return {
       rules,
       model: {},
-      isFormComplete: false,
+      costsModel: {},
+      FACILITY_TYPE_INFO_TXT: 'This is a placeholder message',
```
[the remark this answers is not in the document]
Comment (vietle-cgi) [reply frontend/src/views/applications/OperatingCostsView.vue]: it is updated

File: frontend/src/components/applications/YearlyFacilityCost.vue [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+    facilityType: {
+      type: Number,
+      default: 0,
+    },
+  },
+  emits: ['update'],
+  data() {
+    return {
+      rules,
+      FACILITY_TYPES,
```
[the remark this answers is not in the document]
Comment (vietle-cgi) [reply frontend/src/components/applications/YearlyFacilityCost.vue]: it is updated

File: frontend/src/components/applications/YearlyFacilityCost.vue [hunk 9]
```diff
@@ -0,0 +1,299 @@
...
+    this.model = {
+      rentLeaseCost: this.currentApplication?.rentLeaseCost ?? 0,
+      mortgageCost: this.currentApplication?.mortgageCost ?? 0,
+      propertyTax: this.currentApplication?.propertyTax ?? 0,
+      strataFee: this.currentApplication?.strataFee ?? 0,
+      applicableFee: this.currentApplication?.applicableFee ?? 0,
+    }
+  },
+  methods: {
+    sanitizeInput(key) {
```
[the remark this answers is not in the document]
Comment (vietle-cgi) [reply frontend/src/components/applications/YearlyFacilityCost.vue]: it is updated. For re-usability, I think I will convert this number input field into a component. I will do it in future sprint.

File: frontend/src/views/applications/OperatingCostsView.vue [hunk 10]
```diff
@@ -88,5 +120,36 @@ export default {
...
+        this.setFailureAlert('Failed to save your application', error)
+      } finally {
+        this.$emit('process', false)
+      }
+    },
+
+    updateModel(updatedModel) {
+      Object.entries(updatedModel)?.forEach(([key, value]) => {
+        this.costsModel[key] = Number(value)
+        this.model[key] = Number(value)
```
[the remark this answers is not in the document]
Comment (vietle-cgi) [reply frontend/src/views/applications/OperatingCostsView.vue]: it is updated

**Review by weskubo-cgi — APPROVED** [review]

**Pull request merged**